### PR TITLE
add s2i metrics to exporter

### PR DESCRIPTION
**Pull request by soulseen**
Signed-off-by: soulseen <sunzhu@yunify.com>

### CODE REVIEW

File: pkg/metrics/metrics.go [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+	}, []string{"namespace"})
+
+	S2irunUnknown = prometheus.NewGaugeVec(prometheus.GaugeOpts{
+		Subsystem: s2iSubsystem,
+		Name:      "s2irun_unknown",
+		Help:      "Number of s2irun which status unknown",
+	}, []string{"namespace"})
+
+	S2ibuilderCreated = prometheus.NewGaugeVec(prometheus.GaugeOpts{
+		Subsystem: s2iSubsystem,
```
Comment (runzexia) [inline pkg/metrics/metrics.go]: S2iBuilder

File: pkg/metrics/metrics.go [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+		Help:      "Number of s2irun failed",
+	}, []string{"namespace"})
+
+	S2irunSucceed = prometheus.NewGaugeVec(prometheus.GaugeOpts{
+		Subsystem: s2iSubsystem,
+		Name:      "s2irun_succeed",
+		Help:      "Number of s2irun succeed",
+	}, []string{"namespace"})
+
+	S2irunUnknown = prometheus.NewGaugeVec(prometheus.GaugeOpts{
```
Comment (runzexia) [inline pkg/metrics/metrics.go]: S2iRun

File: pkg/metrics/metrics.go [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+	metrics.Registry.MustRegister(S2irunUnknown)
+	metrics.Registry.MustRegister(S2ibuilderCreated)
+}
+
+func CollectS2iMetrics(k8sclient client.Client) {
+	var wg sync.WaitGroup
+	var err error
+
+	go func() {
+		wg.Add(1)
```
Comment (runzexia) [inline pkg/metrics/metrics.go]: `wg.add should be used before go, using channel here is good`

**Comment by runzexia** [comment]
Add some docs to explain the fields?

**Comment by soulseen** [comment]
> Add some docs to explain the fields?

already have `Help` description.

**Pull request merged**